core/Tracker/GoalManager.php

```diff
@@ -195,6 +207,49 @@ public function detectGoalMatch($goal, Action $action)
         return $action->getActionUrl();
     }
 
+    private function detectNumericGoalMatch($goal, Action $action, VisitProperties $visitProperties)
+    {
+        switch ($goal['match_attribute']) {
+            case 'visit_duration':
+                $valueToMatchAgainst = ((int) $visitProperties->getProperty('visit_total_time')) / 60;
+                break;
+            case 'visit_total_actions':
+                $valueToMatchAgainst = $visitProperties->getProperty('visit_total_actions') ?: 0;
+                $valueToMatchAgainst = $valueToMatchAgainst + 1; // to account for current action
```

Comment:
not all type of actions account for this. So we can't just do `+1`

Reply:
You mean not all `Action` types add to the `visit_total_actions` dimension?

Reply:
Yep see https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/Actions/Columns/VisitTotalActions.php#L55-L105

Reply:
Removed.